Make password hash dummy value configurable

schelcj opened the pull request:
### Problem

The masked password hash dummy value in v4 has changed from v3. In order to maintain consistency it would be nice to make this dummy value configurable so that it will match the current production objects in v3 instances. 

### Solution

Replace the constant dummy value used throughout to a configuration setting in the `default_config.yaml`. Then replace all instances of the constant with a call to `get_setting()`.


```shell
# current object in radb v3 production
$ whois -h whois.radb.net -- -T mntner MAINT-AS237
mntner:     MAINT-AS237
descr:      Merit Network, Inc.
admin-c:    MICHN-ARIN
tech-c:     MICHN-ARIN
upd-to:     routing-notify@merit.edu
mnt-nfy:    routing-notify@merit.edu
auth:       CRYPT-PW HIDDENCRYPTPW
mnt-by:     MAINT-AS237
changed:    ljb@merit.edu 20220201
source:     RADB

# before patching the dummy value
$ whois -h <v4-host> -- -T mntner MAINT-AS237
mntner:         MAINT-AS237
descr:          Merit Network, Inc.
admin-c:        MICHN-ARIN
tech-c:         MICHN-ARIN
upd-to:         routing-notify@merit.edu
mnt-nfy:        routing-notify@merit.edu
auth:           CRYPT-PW DummyValue  # Filtered for security
mnt-by:         MAINT-AS237
changed:        ljb@merit.edu 20220201
source:         RADB

# after patching to make the dummy value configurable
$ whois -h <v4-host> -- -T mntner MAINT-AS237
mntner:         MAINT-AS237
descr:          Merit Network, Inc.
admin-c:        MICHN-ARIN
tech-c:         MICHN-ARIN
upd-to:         routing-notify@merit.edu
mnt-nfy:        routing-notify@merit.edu
auth:           CRYPT-PW HIDDENCRYPTPW
mnt-by:         MAINT-AS237
changed:        ljb@merit.edu 20220201
source:         RADB
``` 

### Testing

Test the configuration change with `pytest` with the `poetry shell`. Tests passing in development.

```shell
$ poetry shell
$ pytest irrd/conf/test_conf.py
``` 

### Notes

* not certain yet if changing in the config requires a full restart or just *SIGUSR1*.
* moved the `# Filtered for security` comment to the default config instead of playing games with the format string.

schelcj committed:
set the default config value for the dummy hash to the existing
constant in a config value then lookup the setting in all places where
the constant was previously used.

schelcj committed:
Merge branch 'irrdnet:main' into make-password-hash-placeholder-configurable

schelcj committed:
No need for the constant just set the default value in the default
config.

schelcj committed:
add the comment back to the default config

schelcj committed:
added configuration setting to the documentation

schelcj committed:
change the `placeholder` name to match the previous constant name.

mxsasha commented:
PR looks good in concept. To fix the linting, run `poe lint` to reformat the code. I can't directly see why the test is failing, I think there may be some issue in detecting whether an object has all dummy hashes or no dummy hashes.

schelcj committed:
running the linter to cleanup things a bit

schelcj committed:
Merge branch 'main' into make-password-hash-placeholder-configurable

schelcj committed:
Merge branch 'main' into make-password-hash-placeholder-configurable

schelcj committed:
more linting cleanup

schelcj committed:
merge conflict cleanup

schelcj committed:
having the comment as part of the value throws off auth line detection/cleanup

schelcj committed:
removed the comment on dummy values. setting in the config breaks
multiple tests. if the comment is desired can revert this one and try
something else.

schelcj commented:
> PR looks good in concept. To fix the linting, run `poe lint` to reformat the code. I can't directly see why the test is failing, I think there may be some issue in detecting whether an object has all dummy hashes or no dummy hashes.

Ok, all the tests were failing because I moved the comment into the config. I removed the comment just to get to a passing state. No idea how anyone feels about that comment, seems redundant but if you want to keep it I'll figure something out.

Still failing the linting test but when I run the same command I get no changes applied, so not sure what's going on there.

I did run `poe lint` and it changed quite a bit unrelated to this PR. Do you want me to revert those changes?

schelcj committed:
Merge branch 'irrdnet:main' into make-password-hash-placeholder-configurable

schelcj commented:
This became a bit of a mess with linting chanes and this isn't really relevant anymore. I'm closing this one out.
